matter_server/CHANGELOG.md

```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 6.4.0
+
+- Bump Python Matter Server to [6.4.0](https://github.com/home-assistant-libs/python-matter-server/releases/tag/6.4.0)
```

Comment:
We didn't bump the server version in build.yaml.

Reply:
Dang, yeah the `--ota-provider-dir` was already present in the previous version, so I did not notice :see_no_evil: 

That also explains this observation https://github.com/home-assistant/core/issues/123086#issuecomment-2269034517:

> In addition, I've just installed the 6.4.0 Matter Server update and both the Aqara P2s are still prompting to update to 1.0.1.1.

The `softwareVersionValid` check was something which got added in 6.4.0.

I'll create a new PR to bump it.

We need to automate this. This is not sustainable. :cry: